### DISCUSSION TITLE
Error when authenticating with Slack

### DISCUSSION THREAD

**Discussion by niflaot** [outline/outline#4584] [Self Hosting]
Yesterday my authentication gateway from slack started failing with the same configuration it started working well. This is the result of the log when we perform the login.

`
{
   "error":"State not return in OAuth flow",
   "stack":"BadRequestError: State not return in OAuth flow\n    at OAuthStateMismatchError (/opt/outline/build/server/errors.js:132:34)\n    at StateStore.verify (/opt/outline/build/server/utils/passport.js:55:61)\n    at Strategy.OAuth2Strategy.authenticate (/opt/outline/node_modules/passport-oauth2/lib/strategy.js:222:26)\n    at attempt (/opt/outline/node_modules/passport/lib/middleware/authenticate.js:369:16)\n    at authenticate (/opt/outline/node_modules/passport/lib/middleware/authenticate.js:370:7)\n    at /opt/outline/node_modules/@outlinewiki/koa-passport/lib/framework/koa.js:194:7\n    at new Promise (<anonymous>)\n    at /opt/outline/node_modules/@outlinewiki/koa-passport/lib/framework/koa.js:193:12\n    at /opt/outline/node_modules/@outlinewiki/koa-passport/lib/framework/koa.js:143:7\n    at new Promise (<anonymous>)\n    at passportAuthenticate (/opt/outline/node_modules/@outlinewiki/koa-passport/lib/framework/koa.js:107:15)\n    at passportMiddleware (/opt/outline/build/server/middlewares/passport.js:83:7)\n    at /opt/outline/node_modules/dd-trace/packages/datadog-instrumentations/src/koa.js:87:57\n    at passportMiddleware (/opt/outline/node_modules/dd-trace/packages/datadog-shimmer/src/shimmer.js:22:21)\n    at dispatch (/opt/outline/node_modules/koa-router/node_modules/koa-compose/index.js:44:32)\n    at next (/opt/outline/node_modules/koa-router/node_modules/koa-compose/index.js:45:18)\n    at /opt/outline/node_modules/koa-router/lib/router.js:346:16\n    at dispatch (/opt/outline/node_modules/koa-router/node_modules/koa-compose/index.js:44:32)\n    at next (/opt/outline/node_modules/koa-router/node_modules/koa-compose/index.js:45:18)\n    at rateLimiterMiddleware (/opt/outline/build/server/middlewares/rateLimiter.js:33:14)\n    at /opt/outline/node_modules/dd-trace/packages/datadog-instrumentations/src/koa.js:87:57\n    at rateLimiterMiddleware (/opt/outline/node_modules/dd-trace/packages/datadog-shimmer/src/shimmer.js:22:21)",
   "level":"error",
   "message":"Error during authentication"
}
`

**To Reproduce**
1. Configure a new Outline install.
2. Start with Slack as authentication provider.
3. Create some documents for around a week or two weeks.

**Expected behavior**
Slack authentication working.

**Screenshots**
[https://i.imgur.com/T01Gi3z.png](url)

**Outline:**
- Install: Self Hosted - Docker
- Version: [0.67.0-pre.1](https://hub.docker.com/layers/outlinewiki/outline/0.67.0-pre.1/images/sha256-437fe82d47126f331587b8f2cd0995ac2c7bcd343f1d4aa18db2ad15b9f6071c?context=explore)

**Desktop:**
 - OS: MacOS
 - Browser Chrome
 - Version 108

**Comment by nickk024** [comment]
I can resolve this issue on my own build build by configuring my reverse proxy as follows. I use Caddy, but there are a lot of guides using nginx around that I tried to adapt from.

xxx.com {
        reverse_proxy * x.x.x.x:3000 {
                header_up Host xxx.com
                header_up X-Forwarded-Proto https
}
}

This has fixed that error with Slack and Google logins. I still am having trouble logging in with regular personal gmail accounts, but am still unclear on what settings I need to actually use!
